### PR TITLE
Adding more Analytics rules

### PR DESCRIPTION
**Pull request by jctaillandier**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by jctaillandier** [comment]
I get:
```
                    Errors: The name 'risk_reasons_s' does not refer to any known column, table, variable or function., Code: 'KS142', Severity: 'Error', Location: '20..34'
```
Although `.script/tests/KqlvalidationsTests/CustomTables/Firework_CL.json` does contain `risk_score_d` as a declared variable.

**Comment by v-prasadboke** [comment]
Hello @jctaillandier please change the unique id of analytic rules.
All the analytic rules contains the same unique id

**Comment by v-prasadboke** [comment]
Hello @jctaillandier Please update your sample data 
' risk_reasons_cl ' is working fine for me 

Screenshot (37)

**Comment by v-prasadboke** [comment]
The Firework_cl table you have uploded does not contain the colum ' risk_reasons_s ' whereas it contains ' risk_source_d ' that is the reason ' risk_source_d ' is working fine

File: Solutions/Flare/Analytic Rules/FlareInfectedDevice.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+relevantTechniques:
+  - T1555
+query: |
+  Firework_CL
+  | where category_name_s contains "Infected Device" or source_s "genesis_market" and (risk_score_d == "3" or risk_score_d == "4" or risk_score_d == "5")
```
Comment (v-prasadboke) [inline Solutions/Flare/Analytic Rules/FlareInfectedDevice.yaml:21]: please correct this query 
It should be [ source_s contains "geneis_market" ]   **OR**  [ source_s == "geneis_market" ]

**Comment by jctaillandier** [comment]
@v-prasadboke I did but now get errors saying I change the ID value of my analytics rule

**Comment by v-prasadboke** [comment]
> I keep getting:
> 
> ```
>                     Errors: The name 'risk_reasons_s' does not refer to any known column, table, variable or function., Code: 'KS142', Severity: 'Error', Location: '20..34'
> ```
> 
> Although `.script/tests/KqlvalidationsTests/CustomTables/Firework_CL.json` does contain `risk_score_d` as a declared variable.

is this working now or you are getting the same error message

File: Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-id: 9cb7c337-f170-4af6-b0e8-b6b7552d762d
+id: 9cb7c337-f171-4af6-b0e8-b6b7552d762d
```
Comment (v-prasadboke) [inline Solutions/Flare/Analytic Rules/FlareCredentialLeaks.yaml:1]: Hello @jctaillandier please revert this change to the old.
Skip  and KQL validation is failing due to this one

**Comment by jctaillandier** [comment]
> > Although `.script/tests/KqlvalidationsTests/CustomTables/Firework_CL.json` does contain `risk_score_d` as a declared variable.
> 
> is this working now or you are getting the same error message

This one seems fine

**Comment by jctaillandier** [comment]
Seems good to go

File: Solutions/CyberArkEPM/Package/mainTemplate.json
Comment (v-prasadboke) [inline Solutions/CyberArkEPM/Package/mainTemplate.json:1]: Hello @jctaillandier why the maintemplate of CyberArkEPM is removed.
Any specific reason for this

File: Solutions/DNS Domain Solution for Microsoft Sentinel/Package/mainTemplate.json
Comment (v-prasadboke) [inline Solutions/DNS Domain Solution for Microsoft Sentinel/Package/mainTemplate.json:1]: Maintemplate of This solution is modified as well.
Any specific reason

**Review by v-prasadboke — CHANGES_REQUESTED** [review]

File: Solutions/CyberArkEPM/Package/mainTemplate.json
Comment (v-prasadboke) [inline Solutions/CyberArkEPM/Package/mainTemplate.json:1]: you can discard the changes for this file or just remove it

**Review by v-prasadboke — APPROVED** [review]

**Comment by v-prasadboke** [comment]
All looks good

**Comment by v-prasadboke** [comment]
Hello @jctaillandier please discard the changes from ' Create-Azure-Sentinel-Solution/v2/input ' folder as well as from ' Create-Azure-Sentinel-Solution/input '

**Comment by jctaillandier** [comment]
Done @v-prasadboke

**Comment by v-prasadboke** [comment]
Hello @jctaillandier is there any changes in the zip 2.0.2

**Comment by v-prasadboke** [comment]
Please update the version of analytic rule FlareDetectionLeaks.yaml

**Comment by v-prasadboke** [comment]
Hello @jctaillandier please update the api version of ' Microsoft.Resources/templateSpecs ' ' 2021-05-01 ' to  ' 2022-02-01 '.
And please update the version of detection " FlareDetectionLeaks.yaml " as well,
Thank you.

**Comment by v-prasadboke** [comment]
Going through this

**Comment by v-prasadboke** [comment]
Hello @jctaillandier have you repackaged the solution after incrementing the version of flarecredentailleaks

**Comment by jctaillandier** [comment]
No, I can do it. Is there a list I should look at of things to do ?

**Comment by v-prasadboke** [comment]
Sorry @jctaillandier for the changes i have requested multiple times,
But we have to go through every instance so that the solution doesnt crash or contain any bugs or issues which would lead to extra work.

**Comment by v-prasadboke** [comment]
Hello @jctaillandier please update the api version of ' Microsoft.Resources/templateSpecs ' ' 2021-05-01 ' to ' 2022-02-01 '.

**Comment by v-prasadboke** [comment]
And please remove the files from v2/input folder

**Comment by jctaillandier** [comment]
done..

**Comment by v-prasadboke** [comment]
> done..

Ok going through this

**Comment by v-prasadboke** [comment]
Hello @jctaillandier Delete the zip version 2.1.0 and create a new one with 2.0.4 and this zip should only include createui and maintemplate,
Thank you

**Comment by v-prasadboke** [comment]
And please discard the changes the v2/input folder

**Comment by jctaillandier** [comment]
I want this version to be 2.1.0, is that ok ?

Could you make a list of things I need to do and share it with me once. We have been going at this for weeks with 1 change at the time.

**Comment by jctaillandier** [comment]
Thanks. 

I just fixed the package, but I do see data_new_leaks_s in my sample (see screenshot). Not sure where else you mean?

Screenshot at 2023-05-08 12-27-33

**Comment by v-prasadboke** [comment]
Hello @jctaillandier this is the sample data I'm referring,
https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Sample%20Data/Custom/Firework_CL.json
This sample data does not contain column - data_new_leaks_s

**Comment by v-prasadboke** [comment]
Hello @jctaillandier this is the sample data I'm referring,
https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Sample%20Data/Custom/Firework_CL.json
This sample data does not contain column - data_new_leaks_s

**Comment by v-prasadboke** [comment]
Hello @jctaillandier this is the sample data I'm referring,
https://raw.githubusercontent.com/Azure/Azure-Sentinel/master/Sample%20Data/Custom/Firework_CL.json
This sample data does not contain column - data_new_leaks_s

**Comment by jctaillandier** [comment]
Done @v-prasadboke

**Comment by v-prasadboke** [comment]
Sorry @jctaillandier but i dont see any commit pushed by you on the branch, Can you please push it again. Thanks, Prasad

**Comment by jctaillandier** [comment]
@v-prasadboke Should be good now

**Review by v-prasadboke — APPROVED** [review]
All looks good

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**